### DISCUSSION TITLE
Error issue when configuring ElastAlert2 with kafka

### DISCUSSION THREAD

**Discussion by Nghia10a5** [jertel/elastalert2#1343] [Q&A] [answered]
Hello everyone,
I am configuring Alert for ELK specifically ElastAlert2,
Because I have a push notification system that will read logs from the kafka topic.
My goal: Want to install ElastAlert2, configure to push error log to kafka topic. Then my push notification system will read Kafka to send Alerts to my workchat.

I'm getting the error:

1.I want to know if ElastAlert2 supports Kafka?
In the requirements.txt file, the kafka-python package is not found.
2.I am getting an error when running the command
python3.9 -m elastalert.elastalert --verbose --rule rules/example_frequency.yaml
Not sure where the error is. Can anyone help me with this problem? I will send detailed error information
Thanks team!

**Comment by nsano-rururu** [comment]
kafka does not support

**Reply by Nghia10a5** [reply]
Hi @nsano-rururu ,
Are you sure ElastAlert2 doesn't support kafka?

**Reply by nsano-rururu** [reply]
There are currently no plans for support.

**Reply by Nghia10a5** [reply]
Thank you!
I am assured of what you say
I will switch to settings to warn about emails
Hopefully it will work, what do you think?

**Reply by nsano-rururu** [reply, marked as answer]
Email doesn't have much maintenance, so I don't recommend it.
I remember that setting gmail or microsoft 365 for smtp_host, smtp_port, from_addr, etc. didn't work well.
https://github.com/johnsusek/praeco/issues/245

I think it's better to use notifications for Slack, Microsoft Teams, Telegram, Discord, etc.

**Comment by nsano-rururu** [comment]
I haven't tried it, but I found the following
https://github.com/0xStormEye/elastalert_kafka

**Reply by Nghia10a5** [reply]
I also found this link
But installing it is quite difficult
But I'm not sure I can install it

**Reply by Nghia10a5** [reply]
Hi @nsano-rururu ,
This link seems like kafka only supports ElastAlert 1
I see this archive is 4-5 years old
Not very accurate.
